contrib/backporting/check-stable

```diff
@@ -141,9 +141,9 @@ generate_commit_list_for_pr () {
   # Use GitHub to determine the number of commits to list, but then query
   # the branch directly to determine the canonical order of the commits.
   for commit in $(git rev-list --reverse -$n_commits $branch); do
-    patchid="$(git show $commit | git patch-id)"
+    patchid="$(git log -n1 --pretty=format:"%ae%at" $commit)"
```

Comment:
I like the overall idea. But I am a bit nervous that `%at` could lead to non-unique IDs because of the low timestamp resolution:

```
git log --pretty=format:"%ae%at" | sort | uniq -c | sort -n
```

Shows quite a few non-unique commits. I think this is is caused by people using `git rebase --reset-author-date` when cleaning up the git history.

Reply:
Note the identification of commits also uses the commit title, so the command should be:
```
git log --pretty=format:"%ae%at%s" | sort | uniq -c | sort -n
```
There are a few duplicates (highest count is two), mostly from the early days of Cilium, but they seem to be commits that were reverted (sometimes by mistake) and reapplied a second time.

Reply:
Maybe we can also use the commit dates. We have the time the PRs are merged and that should be within the same minute (?) as the master commits' dates.

Reply:
Ah, I totally missed the fact that we also used the commit title. Then I think then it should be reasonably safe, and it already is a clear improvement over the status quo.

This is an instance where I think it's totally fine to not cover all few edge cases and instead bail out if we detect that an ID is not unique. In the end, the consequence of the script failing to correlate is just that the user will have to look up the ID themselves in the GitHub UI.